annif/rest.py

```diff
@@ -214,3 +215,72 @@ def learn(
         return server_error(err)
 
     return None, 204
+
+
+def _reconcile(project_id: str, query: dict[str, Any]) -> dict[str, Any]:
```

Comment:
The return type hint `dict` doesn't seem to match the actual returned type, which is a list of dicts.

(found by SonarCloud)